akka-docs/src/test/java/jdocs/typed/tutorial_4/DeviceManager.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.typed.tutorial_4;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static jdocs.typed.tutorial_4.DeviceManagerProtocol.*;
+
+//#device-manager-full
+public class DeviceManager extends AbstractBehavior<DeviceManagerMessage> {
+
+  public static Behavior<DeviceManagerMessage> behavior() {
```

Comment:
For Java I named the factory method `behavior`, but perhaps `create` is better?

For Scala I used `apply` in the companion object. I know there have been different opinions about that. wdyt?